Remove tx_history and ledger_header in API version 2 (#4759)
XRPLF/rippled: API-CHANGELOG.md

Since `api_version` 2 is in "beta", breaking changes to V2 can currently be made at any time.

> #### Retired methods

Author: Thanks !

In API version 2, the following methods are no longer available:

> - `tx_history` - please use other appropriate method, e.g. `tx` or `account_tx` or `ledger` with `transactions` field set to `true`
> - `ledger_header` - please use `ledger` with `binary` field set to `true`

Reviewer: The point of `tx_history` is to get a list of recently applied transactions. The `account_tx` and `ledger` methods are the next best thing for that effect. (I did find one narrow case where tx_history is actually more useful, which is on a chain with very little activity, it can give you recent transactions with fewer steps/foreknowledge than the other two. That need is very niche, mostly served by the ledger command, and less useful because of tx_history's non-standard format, so it's still fine to deprecate IMO.) The `tx` method just looks up an individual transaction if you already know its hash, which is different enough that I think we should not suggest it here.

The `binary` note on `ledger_header` is a red herring; both `ledger` and `ledger_header` return the same (non-binary) info by default. Binary support was the last thing on the `ledger_header` side of the venn diagram, but now the `ledger` command is a superset of the functionality.

```suggestion
- `tx_history` - Instead, use other methods such as `account_tx` or `ledger` with the `transactions` field set to `true`.
- `ledger_header` - Instead, use the `ledger` method.
```

P.S. The word "please" can convey a lot of different tones, not all of them good, especially if we only include it in some places and not others. For the changelog, I think it's best to omit the word "please" from these types of notes.

Author: Thanks !
